integrations/octopus/.port/resources/port-app-config.yml

```diff
@@ -0,0 +1,88 @@
+createMissingRelatedEntities: true
+deleteDependentEntities: true
+resources:
+  - kind: space
+    selector:
+      query: "true"
+    port:
+      entity:
+        mappings:
+          identifier: .Id
+          title: .Name
+          blueprint: '"octopusSpace"'
+          properties:
+            url: env.OCEAN__INTEGRATION__CONFIG__SERVER_URL + "/app#/" + .Id
+            description: .Description
+  - kind: project
+    selector:
+      query: "true"
+    port:
+      entity:
+        mappings:
+          identifier: .Id
+          title: .Name
+          blueprint: '"octopusProject"'
+          properties:
+            url: env.OCEAN__INTEGRATION__CONFIG__SERVER_URL + "/app#/" + .SpaceId + "/projects/" + .Id
+            description: .Description
+            isDisabled: .IsDisabled
+            tenantedDeploymentMode: .TenantedDeploymentMode
+          relations:
+            space: .SpaceId
+  - kind: release
+    selector:
+      query: "true"
+    port:
+      entity:
+        mappings:
+          identifier: .Id
+          title: ".ProjectId + \"(\" + .Version + \")\""
+          blueprint: '"octopusRelease"'
+          properties:
+            version: .Version
+            assembledDate: .Assembled
+            channelId: .ChannelId
+            releaseNotes: .ReleaseNotes
+            url: env.OCEAN__INTEGRATION__CONFIG__SERVER_URL + "/app#/" + .SpaceId + "/releases/" + .Id
+          relations:
+            project: .ProjectId
+  - kind: deployment
+    selector:
+      query: "true"
+    port:
+      entity:
+        mappings:
+          identifier: .Id
+          title: .Name
+          blueprint: '"octopusDeployment"'
```

Comment:
is this `id` is actually unique throughout all spaces and projects? if not we need to make sure we concatenate between them

Reply:
Yes, it should be unique. While the id is autoincrement-based, it comes prepended with the resource name like "Spaces-1", "Machines-1" etc. It is also unique across various projects so the same id is not repeated for two resources in different spaces or project.